mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -336,4 +339,36 @@
      */
     List<MailboxPath> list(MailboxSession session) throws MailboxException;
 
-}
+    /**
+     * Return all mailbox's annotation as the {@link List} of {@link MailboxAnnotation}
+     * 
+     * @param mailboxPath   the current mailbox
+     * @param session       the current session
+     * @return              List<MailboxAnnotation>
+     * @throws MailboxException
+     */
+    List<MailboxAnnotation> getAnnotations(MailboxPath mailboxPath, MailboxSession session) throws MailboxException;
+
+    /**
+     * Get all mailbox's annotation by the list of the keys
```

Comment:
Actually, when reading code using this API, it way easier to read getAllAnnotations() and getAnnotationsByKeys() than getAnnotations() with 2 or 3 parameters. Don't you agree ?